src/submit/codeforces.rs

```diff
@@ -0,0 +1,408 @@
+    let problem_id = url.split('/').nth(6).unwrap();
+    let mut client = client::WebClient::new();
+    if let Err(_) = client.login() {
+        return;
+    }
+    if let Ok(id) = client.submit(contest_id, problem_id, is_gym) {
+        let mut last_len = 0;
+        loop {
+            if let Ok(body) = client.get_url(&format!("https://codeforces.com/{}/{}/submission/{}", if is_gym { "gym" } else { "contest" }, contest_id, id)) {
+                let outcome = Regex::new(r#"<span class='verdict-(\w*)'>(([^<]|<s)*)</span>"#).unwrap();
+                let caps = outcome.captures(&body).unwrap();
+                let outcome_type = caps[1].to_string();
+                let outcome = caps[2].to_string().replace("<span class=\"verdict-format-judged\">", "");
+                for _ in 0..last_len {
+                    print!("{}", 8u8 as char);
+                }
+                for _ in 0..last_len {
+                    print!(" ");
+                }
+                for _ in 0..last_len {
+                    print!("{}", 8u8 as char);
+                }
+                last_len = match outcome_type.as_str() {
+                    "waiting" => pending(&format!("{}", outcome)),
+                    "accepted" => {
+                        success(&format!("{}", outcome));
+                        return;
+                    }
+                    "rejected" => {
+                        failure(&format!("{}", outcome));
+                        return;
+                    }
+                    _ => {
+                        failure(&format!("Unknown: {}", outcome));
+                        return;
+                    }
+                };
+                std::thread::sleep(std::time::Duration::from_secs(1));
+            } else {
+                return;
+            }
+        }
+    }
+}
+
+fn success(s: &str) -> usize {
+    let mut stdout = std::io::stdout();
+    let _ = execute!(stdout, SetForegroundColor(Color::Green));
+    println!("{s}");
+    let _ = execute!(stdout, ResetColor);
+    s.len()
+}
+
+fn failure(s: &str) -> usize {
+    let mut stdout = std::io::stdout();
+    let _ = execute!(stdout, SetForegroundColor(Color::Red));
+    println!("{s}");
+    let _ = execute!(stdout, ResetColor);
+    s.len()
+}
+
+fn pending(s: &str) -> usize {
+    let mut stdout = std::io::stdout();
+    let _ = execute!(stdout, SetForegroundColor(Color::Yellow));
+    print!("{s}");
+    let _ = execute!(stdout, ResetColor);
+    s.len()
+}
+
+// This was mostly written by woshiluo, I just fixed protocol due to codeforces changes
+mod client {
+    use regex::Regex;
+    use std::sync::Arc;
+    use std::fmt;
+    use std::fs::{create_dir_all, File};
+    use std::io::Read;
+    use dialoguer::console::Term;
+    use dialoguer::{Input, Password};
+    use dialoguer::theme::ColorfulTheme;
+    use crate::submit::codeforces::{failure, success};
+
+
+    const SESSION_FILE: &str = "~/.config/rust-competitive-helper/codeforces.session";
```

Comment:
Does it really work? I thought `~` is just a bash feature and shouldn't work in Rust.